### PR TITLE
feat: [CDS-59522]: support shared paths

### PR DESCRIPTION
**Pull request by abhinav-harness**
* feat: [PIE-6426]: Update Order

* feat: [PIE-6426]: Update patch version## Describe your changes

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[PIE-6426]: https://harness.atlassian.net/browse/PIE-6426?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PIE-6426]: https://harness.atlassian.net/browse/PIE-6426?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)

### CODE REVIEW

**Review by rohitk-h — APPROVED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 11 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by sainath-b — APPROVED** [review]

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 11 file(s) to review. Waiting on team(s): del-team.

**Comment by abhinavcode** [comment]
trigger ut0

**Comment by abhinavcode** [comment]
trigger ut2

**Comment by abhinavcode** [comment]
trigger ut5

**Comment by abhinavcode** [comment]
trigger ut8

**Comment by abhinavcode** [comment]
trigger ut9

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 11 file(s) to review. Waiting on team(s): del-team.

**Review by RajBaviskar — APPROVED** [review]

**Comment by abhinavcode** [comment]
trigger messagecheck

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 18 file(s) to review. 1 of 2 approvals obtained. 

___
*[`879-pipeline-ci-commons/src/main/java/io/harness/ci/serializer/BaseRunTimeInputHandler.java` line 19 at r2](https://reviewable.io/reviews/harness/harness-core/46813#-NTzZf_CEuKnwacLMu3k:-NTzZf_CEuKnwacLMu3l:bs416ji) ([raw file](https://github.com/harness/harness-core/blob/7f933d6209d8837368f6b19feac2ae1daf2f6baa/879-pipeline-ci-commons/src/main/java/io/harness/ci/serializer/BaseRunTimeInputHandler.java#L19)):*
> ```Java
>   public static SecretRefData resolveSecretRefWithDefaultValue(String fieldName, String stepType, String stepIdentifier,
>       ParameterField<SecretRefData> parameterField, boolean isMandatory) {
>     if (parameterField == null || parameterField.getValue() == null) {
> ```

nit: can use ParameterField.IsNull() function.

**Review by abhinav-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 18 file(s) to review. 1 of 2 approvals obtained. 

___
*[`879-pipeline-ci-commons/src/main/java/io/harness/ci/serializer/BaseRunTimeInputHandler.java` line 19 at r2](https://reviewable.io/reviews/harness/harness-core/46813#-NTzZf_CEuKnwacLMu3k:-NTzed-XDanotFd__ECR:bsq248h) ([raw file](https://github.com/harness/harness-core/blob/7f933d6209d8837368f6b19feac2ae1daf2f6baa/879-pipeline-ci-commons/src/main/java/io/harness/ci/serializer/BaseRunTimeInputHandler.java#L19)):*
<details><summary><i>Previously, rohitk-h (Rohit Karelia) wrote…</i></summary><blockquote>

nit: can use ParameterField.IsNull() function. 
</blockquote></details>

Used 0 brain same code as CI :smile:

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 18 file(s) to review. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — DISMISSED** [review]
> Reviewed 7 of 11 files at r1, 10 of 11 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: :shipit: complete! ✓

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java [hunk 1]
```diff
@@ -66,30 +70,43 @@ public PluginCreationResponse getPluginInfo(PluginCreationRequest request) {
...
+            .addAllPortUsed(containerDefinitionInfo.getPorts())
+            .setTotalPortUsedDetails(PortDetails.newBuilder().addAllUsedPorts(ports).build())
+            .setResource(getPluginContainerResources(containerDefinitionInfo))
+            .addAllSecretVariable(secretVariables);
+
+    if (containerDefinitionInfo.getRunAsUser() != null) {
+      pluginDetailsBuilder.setRunAsUser(containerDefinitionInfo.getRunAsUser());
+    }
+
+    if (!(ciAbstractStepNode instanceof BackgroundStepNodeV1 || ciAbstractStepNode instanceof BackgroundStepNode)) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:96]: why do we have to add this if condition ?

**Comment by abhinav-harness** [comment]
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java` line 96 at r4](https://reviewable.io/reviews/harness/harness-core/46813#gh-1178669064:-NU0kSxOEg2SVrWULzQM:be3zjr7) ([raw file](https://github.com/harness/harness-core/blob/3a9fa80bde57a2ae0579ee15c68fb0817032cf06/332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java#L96)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

why do we have to add this if condition ?
</blockquote></details>

Bg step is not instance of plugincompatible step - we have 2 step supported today so having this if condition

**Review by harshjain12 — APPROVED** [review]

**Review by harshjain12 — DISMISSED** [review]

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 1 file(s) to review. 1. unresolved discussions.

**Comment by abhinav-harness** [comment]
trigger ut2

**Comment by abhinav-harness** [comment]
trigger ut7

**Comment by abhinav-harness** [comment]
trigger ut6

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 5 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 5 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by harshjain12 — APPROVED** [review]

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, 4 of 4 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46813)* status: 1. unresolved discussions.

**Pull request merged**